geoh5py/ui_json/forms.py

```diff
@@ -453,23 +493,34 @@ def __init__(self, name, value=None, **kwargs):
             ],
         )
         self._is_value = BoolParameter("is_value")
-        self._property = UUIDParameter("property")
+        self._property = TypeRestrictedParameter(
+            "property", [self._data_type_string_to_type(data_type)]
+        )
         value = NumericParameter("value", value=value)
-        super().__init__(name, value=value, **kwargs)
+        super().__init__(name, value=value, data_type=data_type, **kwargs)
 
     @property
     def value(self):
         """Form value is value of property when is_value is False."""
-        val = self.property
-        if self.is_value:
+        val = self.property  # type: ignore # pylint: disable=no-member
+        if self.is_value:  # type: ignore # pylint: disable=no-member

```

Comment:
How does this work? `is_value` is a BoolParameter, so this would always pass.

Reply:
The private attribute `self._is_value` is a BoolParameter, but I'm accessing the underlying value here with the public name through the ValueAccess descriptor.